shanoir-ng-import/src/main/java/org/shanoir/ng/importer/ImporterApiController.java

```diff
@@ -781,4 +801,141 @@ public ResponseEntity<ByteArrayResource> getDicomImage(@ApiParam(value = "path",
+			@ApiParam(value = "studyId", required = true) @PathVariable("studyId") Long studyId,
+			@ApiParam(value = "studyName", required = true) @PathVariable("studyName") String studyName,
+			@ApiParam(value = "studyCardId", required = true) @PathVariable("studyCardId") Long studyCardId,
+			@ApiParam(value = "centerId", required = true) @PathVariable("centerId") Long centerId,
+			@ApiParam(value = "converterId", required = true) @PathVariable("converterId") Long converterId) throws RestServiceException {
+		// STEP 1: Unzip file
+		if (dicomZipFile == null || !ImportUtils.isZipFile(dicomZipFile)) {
+			throw new RestServiceException(
+					new ErrorModel(HttpStatus.UNPROCESSABLE_ENTITY.value(), WRONG_CONTENT_FILE_UPLOAD, null));
+		}
+
+		ImportJob job = null;
+
+		try {
+			File userImportDir = ImportUtils.getUserImportDir(importDir);			
+			File tempFile = ImportUtils.saveTempFile(userImportDir, dicomZipFile);
+			File importJobDir = ImportUtils.saveTempFileCreateFolderAndUnzip(tempFile, dicomZipFile, true);
+
+			// STEP 2: Analyze file structure and verify for mismatch
+			File[] subjectFolders = importJobDir.listFiles();
+			if (subjectFolders == null || subjectFolders.length != 1) {
+				throw new RestServiceException(new ErrorModel(HttpStatus.UNPROCESSABLE_ENTITY.value(),
+						"The zip must contain a single folder named with the desired subject name.", null));
+			}
+			File subjectFolder = subjectFolders[0];
+			File[] examinationsFolders = subjectFolder.listFiles();
+
+			String subjectName = studyName + "_" + subjectFolder.getName();
+			Subject subject = null;
+
+
+			// STEP 4: Iterate over examination folders
+			for (File examFolder : examinationsFolders) {
+				job = null;
+				// Check of it's a folder
+				if (!examFolder.isDirectory()) {
+					throw new RestServiceException(new ErrorModel(HttpStatus.UNPROCESSABLE_ENTITY.value(),
+							"The main subject folder should only contain sub-folders and not single/data files.", null));
+				}
+				File zippedExamFolder = new File (examFolder.getAbsolutePath() + ".zip");
+				zippedExamFolder.createNewFile();
+				Utils.zip(examFolder.getAbsolutePath(), zippedExamFolder.getAbsolutePath());
+				MockMultipartFile mockedFile = new MockMultipartFile(examFolder.getName(), zippedExamFolder.getName(), APPLICATION_ZIP, new FileInputStream(zippedExamFolder));
+
+				// STEP 4.0 "Fake" upload file to get create the dicomdir and temporary folder
+				job = this.uploadDicomZipFile(mockedFile).getBody();
+
+				// Create subject only once.
+				if (subject == null) {
+					Patient pat = job.getPatients().get(0);
+					// Create subject
+					subject = new Subject();
+					subject.setName(subjectName);
+					subject.setBirthDate(pat.getPatientBirthDate());
+					subject.setSex(pat.getPatientSex());
+					// LIVING_HUMAN_BEING by default
+					subject.setImagedObjectCategory(1);
+					// Here the ID is used to carry the study ID
+					subject.setId(studyId);
```

Comment:
Dear Jean-Come, please do not do this here. Sending a studyId in a subjectId is really limit.
Better add two calls, one for create subject and one for subject-study please.